Source/WebKit/UIProcess/WebFrameProxy.cpp

```diff
@@ -428,14 +428,20 @@ void WebFrameProxy::prepareForProvisionalLoadInProcess(WebProcessProxy& process,
     if (isMainFrame())
         return completionHandler();
 
-    Site navigationSite(navigation.currentRequest().url());
     RefPtr page = m_page.get();
+    Site navigationSite(navigation.currentRequest().url());
     // FIXME: Main resource (of main or subframe) request redirects should go straight from the network to UI process so we don't need to make the processes for each domain in a redirect chain. <rdar://116202119>
     RegistrableDomain mainFrameDomain(page->mainFrame()->url());
 
     m_provisionalFrame = nullptr;
     m_provisionalFrame = makeUnique<ProvisionalFrameProxy>(*this, group.ensureProcessForSite(navigationSite, process, page->preferences()));
-    page->websiteDataStore().protectedNetworkProcess()->addAllowedFirstPartyForCookies(process, mainFrameDomain, LoadedWebArchive::No, WTFMove(completionHandler));
+    protectedProcess()->sendWithAsyncReply(Messages::WebPage::GetFrameOwnerPermissionsPolicy(m_frameID), [weakThis = WeakPtr { *this }, page = m_page, process = Ref { process }, frameID = m_frameID, mainFrameDomain = WTFMove(mainFrameDomain), completionHandler = WTFMove(completionHandler)](auto policy) mutable {
+        if (!page)
```

Comment:
We probably want to take a Ref of the page here.

Reply:
This has been removed.